### PR TITLE
Naming bug in frequency domain analysis

### LINKED ISSUE #25776: muon frequency domain analysis FFT names output workspace name bug
### Expected behavior
output workspace from from transforms in muon frequency domain analysis should be named apropreatly so that they can be distinguished.

### Actual behavior
If a transform in performed without imaginary data, it is named in the same way as if it had be done with imaginary data using the last entry put in the imaginary workspace option.

### Steps to reproduce the behavior
1. open mantid plot
1. go to interfaces-->muon-->frequency domain analysis.
1. open EMU run 51341
1. go to transform tab
1. In workspace use `EMU51341; Pair Asym; long; \#1` and the same again in imaginary workspace
1. click Calcualte FFT
1. A new workspace as been created `FFT; Re EMU51341; PAir Asym; long; \#1; Im EMU51341; Pair Asym; long; \#1`. open and check contents.
1. uncheck Imaginary data and click click Calcualte FFT
1. no new workspace has been created, but `FFT; Re EMU51341; PAir Asym; long; \#1; Im EMU51341; Pair Asym; long; \#1` has been overwritten.

### Platforms affected

check with windows.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The FFT's in frequency domain analysis (FDA) would always add a name for the imaginary part of the FFT, even if one was not used. This fixes it.

**To test:**

Open FDA
Load some data
Go to the transform tab
Do a FFT
In the ADS a workspaces will have been added (its in a nested structure) that will reflect the names used for the real and imaginary parts. 
Untick the use imaginary part
Do an FFT
In the ADS a workspaces will have been added (its in a nested structure) that will reflect the names used for the real and imaginary parts (i.e. no imaginary part). 

Fixes #25776. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by Harrietbrown** [comment]
When I try to calculate FFT without imaginary data I get the error: 
'NoneType' object has no attribute 'group'

File: scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py [hunk 1]
```diff
@@ -175,25 +175,26 @@ def calculate_FFT(self):
...
         self.add_fft_workspace_to_ADS(real_workspace_padding_parameters['InputWorkspace'],
                                       imaginary_workspace_padding_parameters['InputWorkspace'],
                                       frequency_domain_workspace)
 
     def add_fft_workspace_to_ADS(self, input_workspace, imaginary_input_workspace, fft_workspace):
         run = re.search('[0-9]+', input_workspace).group()
-        Im_run = re.search('[0-9]+', imaginary_input_workspace).group()
+        Im_run = ""
+        if imaginary_input_workspace is not "":
+             Im_run = re.search('[0-9]+', imaginary_input_workspace).group()
```
Comment (Harrietbrown) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py]:
```suggestion
            Im_run = re.search('[0-9]+', imaginary_input_workspace).group()
```

Comment (AnthonyLim23) [reply scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py]: ah thanks. I will do it later when I fix the unit tests

File: scripts/test/Muon/fft_presenter_context_interaction_test.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
 from Muon.GUI.FrequencyDomainAnalysis.FFT import fft_view_new
 from Muon.GUI.FrequencyDomainAnalysis.FFT import fft_model
 
+from Muon.GUI.Common.utilities.algorithm_utils import convert_to_field
```
Comment (AnthonyLim23) [inline scripts/test/Muon/fft_presenter_context_interaction_test.py]:
```suggestion
```

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
It all functions fine, just make sure the tests are all passing propperly and it will be good to go.

**Comment by robertapplin** [comment]
@AnthonyLim23 The tests have passed!

**Review by Harrietbrown — APPROVED** [review]

**Pull request merged**